### DISCUSSION TITLE
How to append data with custom pagination

### DISCUSSION THREAD

**Discussion by austin43** [urql-graphql/urql#2459] [Q&A]
Hello, I am trying to design my own cursor-based pagination system for my app which is connected to Hasura.  I figured I'd start with the simplePagination that urql provides as a basis and work from there, but one thing confused me when going through the tests.

Page two appends to page one data, which makes sense to me for the resolver: https://github.com/FormidableLabs/urql/blob/main/exchanges/graphcache/src/extras/simplePagination.test.ts#L65

However, I would expect page three to resolve to the same if there was no more data to query, rather than null.  Can anyone explain why this resolves to null? I would think that this would cause your page to go blank as soon as you hit page 3's query in this case?

https://github.com/FormidableLabs/urql/blob/main/exchanges/graphcache/src/extras/simplePagination.test.ts#L74

**Comment by Svarto** [comment]
@austin43 did you solve this and create a cursor based pagination resolver for URQL? I am working on exactly the same thing and would be great to get some inspiration and help - struggling a bit with making sense of the available resolvers in URQL.

**Reply by austin43** [reply]
@Svarto Sorry for the late reply here.  Yes I was able to dig into the resolvers and make my own pagination using simplePagination as a basis.  https://gist.github.com/austin43/ecd8d70e68d0571905f36b40d99fdf29
